crates/vm/vm.rs

```diff
@@ -75,34 +75,172 @@ impl EvmState {
+            if block_header.parent_beacon_block_root.is_some() && spec_id == SpecId::CANCUN {
+                beacon_root_contract_call(state, block_header, spec_id)?;
+            }
+            let mut receipts = Vec::new();
+            let mut cumulative_gas_used = 0;
+
+            let store_wrapper = Arc::new(StoreWrapper {
+                store: state.database().unwrap().clone(),
+                block_hash: block.header.parent_hash,
+            });
+
+            let mut account_updates: Vec<AccountUpdate> = vec![];
+
+            for transaction in block.body.transactions.iter() {
+                let result = execute_tx_levm(transaction, block_header, store_wrapper.clone()).unwrap();
+                cumulative_gas_used += result.gas_used;
+                let receipt = Receipt::new(
+                    transaction.tx_type(),
+                    matches!(result.result, TxResult::Success),
+                    cumulative_gas_used,
+                    // TODO: map our logs to the logs expected by ethereum_rust
+                    vec![],
+                );
+                receipts.push(receipt);
 
-    if let Some(withdrawals) = &block.body.withdrawals {
-        process_withdrawals(state, withdrawals)?;
-    }
+                for (address, account) in result.new_state {
+                    let mut added_storage = HashMap::new();
+
+                    for (key, value) in account.storage {
+                        added_storage.insert(key, value.current_value);
+                    }
+
+                    let code = if account.info.bytecode.is_empty() {
+                        None
+                    } else {
+                        Some(account.info.bytecode.clone())
+                    };
+
+                    let account_update = AccountUpdate {
+                        address,
+                        removed: false,
+                        info: Some(AccountInfo {
+                            code_hash: code_hash(&account.info.bytecode),
+                            balance: account.info.balance,
+                            nonce: account.info.nonce,
+                        }),
+                        code,
+                        added_storage,
+                    };
+
+                    account_updates.push(account_update);
+                }
+            }
+
+            if let Some(withdrawals) = &block.body.withdrawals {
+                process_withdrawals(state, withdrawals)?;
+            }
+
+            Ok((receipts, account_updates))
+        }
+
+        pub fn execute_tx_levm(
+            tx: &Transaction,
+            block_header: &BlockHeader,
+            db: Arc<dyn LevmDatabase>,
+        ) -> Result<TransactionReport, VMError> {
+            dbg!(&tx.tx_type());
+
+            let gas_price: U256 = match tx.tx_type() {
+                TxType::Legacy => tx.gas_price().into(),
+                TxType::EIP2930 => tx.gas_price().into(),
+                TxType::EIP1559 => {
```

Comment:
Nit: These transactions seem to overlap, consider using a '|' for a 'or' match, like in this [example](https://doc.rust-lang.org/rust-by-example/flow_control/match.html).